pgcli/pgexecute.py

```diff
@@ -263,15 +267,14 @@ def run(self, statement, pgspecial=None, on_error=ON_ERROR_RESUME):
                 _logger.error("traceback: %r", traceback.format_exc())
 
                 if (isinstance(e, psycopg2.OperationalError)
-                        or on_error == ON_ERROR_RAISE):
+                        or not exception_formatter):
                     # Always raise operational errors, regardless of on_error
                     # specification
                     raise
 
-                result = click.style(utf8tounicode(str(e)), fg='red')
-                yield None, None, None, result
+                yield exception_formatter(e)
 
-                if on_error == ON_ERROR_STOP:
+                if not on_error_resume:
```

Comment:
I dont' think this break is necessary, since the raise up above will break us out of the loop.